pkgs/by-name/bi/bisq/package.nix

```diff
@@ -0,0 +1,157 @@
+{
+  stdenvNoCC,
+  lib,
+  makeWrapper,
+  fetchurl,
+  makeDesktopItem,
+  copyDesktopItems,
+  imagemagick,
+  openjdk,
+  dpkg,
+  writeScript,
+  bash,
+  tor,
+  zip,
+  gnupg,
+}:
+
+let
+  bisq-launcher =
+    args:
+    writeScript "bisq-launcher" ''
+      #! ${bash}/bin/bash
+
+      # This is just a comment to convince Nix that Tor is a
+      # runtime dependency; The Tor binary is in a *.jar file,
+      # whereas Nix only scans for hashes in uncompressed text.
+      # ${bisq-tor}
+
+      rm -fR $HOME/.local/share/Bisq2/tor
+
+      exec "${openjdk}/bin/java" -Djpackage.app-version=@version@ -classpath @out@/lib/app/desktop-app-launcher.jar:@out@/lib/app/* ${args} bisq.desktop_app_launcher.DesktopAppLauncher "$@"
+    '';
+
+  bisq-tor = writeScript "bisq-tor" ''
+    #! ${bash}/bin/bash
+
+    exec ${tor}/bin/tor "$@"
+  '';
+
+  # Alejandro Garcia's public key was obtained from
+  # https://bisq.network/pubkey/E222AA02.asc
+  publicKey = ./pubkey.asc;
```

Comment:
Please fetch that with fetchurl instead. We have the hash then, so it cannot be manipulated easily.